MNE/disp3D/3DObjects/brain/brainrtconnectivitydatatreeitem.cpp

```diff
@@ -144,14 +142,161 @@ bool BrainRTConnectivityDataTreeItem::init(const MNEForwardSolution& tForwardSol
+
+    list.clear();
+    MetaTreeItem* pItemNetworkMatrix = new MetaTreeItem(MetaTreeItemTypes::NetworkMatrix, "Show network matrix");
+    list << pItemNetworkMatrix;
+    list << new QStandardItem(pItemNetworkMatrix->toolTip());
+    this->appendRow(list);
+
+    //Plot network
+    plotNetwork(pNetworkData, vecEdgeTrehshold);
+
     return true;
 }
 
+
+//*************************************************************************************************************
+
+void BrainRTConnectivityDataTreeItem::onCheckStateChanged(const Qt::CheckState& checkState)
+{
+    this->setVisible(checkState == Qt::Unchecked ? false : true);
+}
+
+
+//*************************************************************************************************************
+
+void BrainRTConnectivityDataTreeItem::setVisible(bool state)
+{
+    for(int i = 0; i < m_lNodes.size(); ++i) {
+        m_lNodes.at(i)->setParent(state ? m_pRenderable3DEntity : Q_NULLPTR);
+    }
+
+    m_pRenderable3DEntity->setParent(state ? m_pParentEntity : Q_NULLPTR);
+}
+
+
+//*************************************************************************************************************
+
+void BrainRTConnectivityDataTreeItem::onNetworkThresholdChanged(const QVector3D& vecThresholds)
+{
+    qDebug()<<"";
+    Network::SPtr pNetwork = this->data(Data3DTreeModelItemRoles::NetworkData).value<Network::SPtr>();
+
+    plotNetwork(pNetwork, vecThresholds);
+}
+
+
+//*************************************************************************************************************
+
+void BrainRTConnectivityDataTreeItem::plotNetwork(QSharedPointer<CONNECTIVITYLIB::Network> pNetworkData, const QVector3D& vecThreshold)
+{
+    //Prepare network visualization
+    QMatrix4x4 m;
+    Qt3DCore::QTransform* transform =  new Qt3DCore::QTransform();
+    m.rotate(180, QVector3D(0.0f, 1.0f, 0.0f));
+    m.rotate(-90, QVector3D(1.0f, 0.0f, 0.0f));
+    transform->setMatrix(m);
+    m_pRenderable3DEntity->addComponent(transform);
+
+    QVector3D pos;
+    QSharedPointer<Qt3DCore::QEntity> sourceSphereEntity;
+    Qt3DExtras::QSphereMesh* sourceSphere;
+    Qt3DExtras::QPhongMaterial* material;
+
+    //Draw network nodes
+    m_lNodes.clear();
+    MatrixX3f tMatVert(pNetworkData->getNodes().size(), 3);
+    MatrixX3f tMatNorm(pNetworkData->getNodes().size(), 3);
+    tMatNorm.setZero();
+    QList<NetworkNode::SPtr> lNetworkNodes = pNetworkData->getNodes();
+
+    for(int i = 0; i < lNetworkNodes.size(); ++i) {
+        pos.setX(lNetworkNodes.at(i)->getVert()(0));
+        pos.setY(lNetworkNodes.at(i)->getVert()(1));
+        pos.setZ(lNetworkNodes.at(i)->getVert()(2));
+
+        tMatVert(i,0) = pos.x();
+        tMatVert(i,1) = pos.y();
+        tMatVert(i,2) = pos.z();
+
+        sourceSphereEntity = QSharedPointer<Qt3DCore::QEntity>(new Qt3DCore::QEntity(m_pRenderable3DEntity));
+
+        sourceSphere = new Qt3DExtras::QSphereMesh();
```

Comment:
sourceSphereEnt is th child of the QEntity. Once this one is destroyed QT will internally handle data management and deleting.